Dicipulo, Earl Geibriel S.

# **Activity 2: Python-Pandas Exercise**

Objectives:
- Understand Python syntax (variables, loops, functions).
- Learn Pandas basics (Series, DataFrames, reading files).
- Perform data cleaning (handling missing values, correcting formats, removing duplicates).
- Apply concepts in a real-world case study.

# Part 1: Hands-on Python & Pandas Basics

1. Install the Pandas library in your environment.

In [ ]:
pip install pandas

2. Import the  pandas package under the name `pd`

In [ ]:
import pandas as pd

3. Print the pandas version

In [3]:
import pandas as pd

print(pd.__version__)

2.2.3


4. Create a variable `x` with the value 10 and a string variable `y` with "Fortes in Fide!"

In [ ]:
x = 10
y = 'Fortes in Fide!'

10
Fortes in Fide!


5. Define a list with numbers `[1, 2, 3, 4, 5]` and a dictionary with keys `name` and `age`

In [ ]:
a = [1, 2, 3, 4, 5]
b = {
    "name",
    "key"
}


6. Write a function `greet(name)` that returns "Magis, (name)"!

In [12]:
def greet(fname):
    print("Magis,", fname, "!")

greet("Earl")

Magis, Earl !


7. Write a Python function that takes a user’s name as input and prints a personalized greeting.

In [ ]:
def greet(username):
    print("Good day to you,", username, "!")

username = input("Enter username:")
greet(username)

Hello Earl !


8. Modify **Number 7** that if the user does not enter a name, it defaults to "Guest".

In [17]:
def greet(username):
    print("Good day to you,", username, "!")

username = "Guest"
greet(username)

Good day to you, Guest !


9. Create a Pandas Series from `[10, 20, 30, 40]`.

In [ ]:
import pandas as pd

x = [10, 20, 30, 40]

myvar = pd.Series(x)
print(myvar)

0    10
1    20
2    30
3    40
dtype: int64


10.  Create a DataFrame with columns `A` and `B`.

In [19]:
import pandas as pd

data = {
    "A": [],
    "B": []
}

dataframe = pd.DataFrame(data)
print(dataframe)

Empty DataFrame
Columns: [A, B]
Index: []


# Part 2: Working with a Dataset 🛥️

1. Load the Titanic dataset from a local file and display the first five rows.

In [22]:
import pandas as pd

pd.options.display.max_rows = 5

dataframe = pd.read_csv('titanic_dataset.csv')
print(dataframe.head())

   PassengerId  Survived  Pclass  \
0          892         0       3   
1          893         1       3   
2          894         0       2   
3          895         0       3   
4          896         1       3   

                                           Name     Sex   Age  SibSp  Parch  \
0                              Kelly, Mr. James    male  34.5      0      0   
1              Wilkes, Mrs. James (Ellen Needs)  female  47.0      1      0   
2                     Myles, Mr. Thomas Francis    male  62.0      0      0   
3                              Wirz, Mr. Albert    male  27.0      0      0   
4  Hirvonen, Mrs. Alexander (Helga E Lindqvist)  female  22.0      1      1   

    Ticket     Fare Cabin Embarked  
0   330911   7.8292   NaN        Q  
1   363272   7.0000   NaN        S  
2   240276   9.6875   NaN        Q  
3   315154   8.6625   NaN        S  
4  3101298  12.2875   NaN        S  


2. Display the dataset's column names, data types.

In [23]:
import pandas as pd

dataframe = pd.read_csv('titanic_dataset.csv')

print(dataframe.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 418 entries, 0 to 417
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  418 non-null    int64  
 1   Survived     418 non-null    int64  
 2   Pclass       418 non-null    int64  
 3   Name         418 non-null    object 
 4   Sex          418 non-null    object 
 5   Age          332 non-null    float64
 6   SibSp        418 non-null    int64  
 7   Parch        418 non-null    int64  
 8   Ticket       418 non-null    object 
 9   Fare         417 non-null    float64
 10  Cabin        91 non-null     object 
 11  Embarked     418 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 39.3+ KB
None


3. Display the dataset's missing values.

In [48]:
import pandas as pd

dataframe = pd.read_csv('titanic_dataset.csv')
print(dataframe.isnull())
print('Number of Missing Values per Column', '\n', dataframe.isna().sum())

     PassengerId  Survived  Pclass   Name    Sex    Age  SibSp  Parch  Ticket  \
0          False     False   False  False  False  False  False  False   False   
1          False     False   False  False  False  False  False  False   False   
..           ...       ...     ...    ...    ...    ...    ...    ...     ...   
416        False     False   False  False  False   True  False  False   False   
417        False     False   False  False  False   True  False  False   False   

      Fare  Cabin  Embarked  
0    False   True     False  
1    False   True     False  
..     ...    ...       ...  
416  False   True     False  
417  False   True     False  

[418 rows x 12 columns]
Number of Missing Values per Column 
 PassengerId      0
Survived         0
              ... 
Cabin          327
Embarked         0
Length: 12, dtype: int64


4. Display the `Name`, `Age`, and `Fare` columns from the dataset. (first 10)

In [28]:
import pandas as pd

dataframe = pd.read_csv('titanic_dataset.csv', usecols=['Name', 'Age', 'Fare'])

print(dataframe)

                                 Name   Age     Fare
0                    Kelly, Mr. James  34.5   7.8292
1    Wilkes, Mrs. James (Ellen Needs)  47.0   7.0000
..                                ...   ...      ...
416               Ware, Mr. Frederick   NaN   8.0500
417          Peter, Master. Michael J   NaN  22.3583

[418 rows x 3 columns]


 5. Print the descriptive statistics of the Titanic dataset.

In [29]:
import pandas as pd

dataframe = pd.read_csv('titanic_dataset.csv')
stats = dataframe.describe()

print(stats)

       PassengerId    Survived     Pclass        Age       SibSp       Parch  \
count       418.00  418.000000  418.00000  332.00000  418.000000  418.000000   
mean       1100.50    0.363636    2.26555   30.27259    0.447368    0.392344   
...            ...         ...        ...        ...         ...         ...   
75%        1204.75    1.000000    3.00000   39.00000    1.000000    0.000000   
max        1309.00    1.000000    3.00000   76.00000    8.000000    9.000000   

             Fare  
count  417.000000  
mean    35.627188  
...           ...  
75%     31.500000  
max    512.329200  

[8 rows x 7 columns]


6. Remove rows with missing values in the `Age` column.

In [30]:
import pandas as pd

dataframe = pd.read_csv('titanic_dataset.csv')
newdataframe = dataframe.dropna(subset=['Age'], inplace=False)

print(newdataframe)

     PassengerId  Survived  Pclass                              Name     Sex  \
0            892         0       3                  Kelly, Mr. James    male   
1            893         1       3  Wilkes, Mrs. James (Ellen Needs)  female   
..           ...       ...     ...                               ...     ...   
414         1306         1       1      Oliva y Ocana, Dona. Fermina  female   
415         1307         0       3      Saether, Mr. Simon Sivertsen    male   

      Age  SibSp  Parch              Ticket      Fare Cabin Embarked  
0    34.5      0      0              330911    7.8292   NaN        Q  
1    47.0      1      0              363272    7.0000   NaN        S  
..    ...    ...    ...                 ...       ...   ...      ...  
414  39.0      0      0            PC 17758  108.9000  C105        C  
415  38.5      0      0  SOTON/O.Q. 3101262    7.2500   NaN        S  

[332 rows x 12 columns]


7. Remove duplicate rows from the dataset.

In [33]:
import pandas as pd

dataframe = pd.read_csv('titanic_dataset.csv')
print(dataframe.drop_duplicates(inplace = False))


     PassengerId  Survived  Pclass                              Name     Sex  \
0            892         0       3                  Kelly, Mr. James    male   
1            893         1       3  Wilkes, Mrs. James (Ellen Needs)  female   
..           ...       ...     ...                               ...     ...   
416         1308         0       3               Ware, Mr. Frederick    male   
417         1309         0       3          Peter, Master. Michael J    male   

      Age  SibSp  Parch  Ticket     Fare Cabin Embarked  
0    34.5      0      0  330911   7.8292   NaN        Q  
1    47.0      1      0  363272   7.0000   NaN        S  
..    ...    ...    ...     ...      ...   ...      ...  
416   NaN      0      0  359309   8.0500   NaN        S  
417   NaN      1      1    2668  22.3583   NaN        C  

[418 rows x 12 columns]


8. Compute and display the correlation matrix of the dataset.

# Part 2: Working with Case Studies

When working on these case studies, **always ensure that your code is properly documented and clearly presented**. Follow these key principles:  

### **1. Always Show Your Code**  
- Every step of data exploration, cleaning, and analysis should include **visible code outputs**.  
- Do not skip showing your process, as transparency is essential for reproducibility.  

### **2. Proper Documentation is Necessary**  
- Use **comments (`#`) in Python** to explain your code clearly.  
- Add **Markdown cells** to describe each step before executing the code.  
- Explain key findings in simple language to make the analysis easy to understand.  

### **3. Use Readable and Organized Code**  
- Follow a **step-by-step approach** to keep the notebook structured.  
- Use **proper variable names** and avoid hardcoding values where possible.

# **Case Study 1: Iris Flower Classification** 🌸  

### **Background**  
A botanical research institute wants to develop an automated system that classifies different species of **iris flowers** based on their **sepal and petal measurements**.  The dataset consists of **150 samples**, labeled as **Setosa, Versicolor, or Virginica**.  

### **Problem Statement**  
Can we use **sepal and petal dimensions** to correctly classify the **species of an iris flower**?  

### **Task Description**  

#### **1. Data Exploration**  
- Load the dataset and display the first few rows.  
- Identify any missing or inconsistent values.  

#### **2. Data Cleaning**  
- Check for missing values and handle them appropriately.  
- Convert categorical species labels into a format suitable for analysis.  

#### **3. Basic Data Analysis**  
- Find the average sepal and petal dimensions for each species.  
- Identify correlations between different flower measurements.  

#### **4. Visualization**  
- Create simple visualizations (e.g., histograms, scatter plots) to understand data distribution.  

#### **5. Insights & Interpretation**  
- Summarize key findings, such as which features best distinguish flower species.  

In [137]:
import pandas as pd
import matplotlib.pyplot as plt

# Load the iris dataset
dataframe = pd.read_csv('iris_dataset.csv')

# Clean Dataset
newdataframe = dataframe.dropna().drop_duplicates() # Removes null and duplicated values from the dataset

# Find the average sepal and petal dimensions from each species
averageDimensions = newdataframe.groupby('species')[['sepal_length', 'sepal_width', 'petal_length', 'petal_width']].mean()
print("Average Dimensions", "\n", averageDimensions, '\n')
print('------------------------------------------------------------------')

# Identify the correlation between different flower measurements
correlation = newdataframe.iloc[:, :-1].corr()
print("Correlation", "\n", correlation, "\n")

# Running the .corr() function on the dataset results in an error because the
# dataset contains a string column in the last part of the dataset and the corr()
# function cannot accept strings.

print('------------------------------------------------------------------')

# Load visualization graph and save it as an image
newdataframe.columns = ["Sepal Length", "Sepal Width", "Petal Length", "Petal Width", "Species"]

correlation.plot()
plt.savefig("correlation_graph.png")
plt.show()




Average Dimensions 
                  sepal_length  sepal_width  petal_length  petal_width
species                                                              
Iris-setosa          5.010417     3.431250      1.462500     0.250000
Iris-versicolor      5.936000     2.770000      4.260000     1.326000
Iris-virginica       6.604082     2.979592      5.561224     2.028571 

------------------------------------------------------------------
Correlation 
               sepal_length  sepal_width  petal_length  petal_width
sepal_length      1.000000    -0.109321      0.871305     0.817058
sepal_width      -0.109321     1.000000     -0.421057    -0.356376
petal_length      0.871305    -0.421057      1.000000     0.961883
petal_width       0.817058    -0.356376      0.961883     1.000000 

------------------------------------------------------------------


C:\Users\Lenovo\AppData\Local\Temp\ipykernel_13124\1843424643.py:30: UserWarning: FigureCanvasAgg is non-interactive, and thus cannot be shown
  plt.show()


Key Findings:

- Iris Setosa has the smallest petal length and petal width, but has the largest sepal width.
- Iris Virginica has the largest sepal length, petal length, and petal width.
- Sepal dimensions are less useful in classification due to its negative correlation.
- Petal dimesions are more useful in classification due io its positive correlation.


# **Case Study 2: Netflix Content Analysis** 🎬  

## **Background**  
Netflix is a leading streaming platform with a vast collection of movies and TV shows. The company wants to analyze its **content library** to understand trends in **genres, release years, and regional distribution**.  

## **Problem Statement**  
How can we use **Netflix’s dataset** to gain insights into content distribution, popular genres, and release trends over time?  

## **Task Description**  

### **1. Data Exploration**  
- Load the dataset and inspect its structure.  
- Identify key columns such as title, genre, release year, and country.  

### **2. Data Cleaning**  
- Check for missing or incorrect values in key columns.  
- Remove duplicates and format the date-related data properly.  

### **3. Basic Data Analysis**  
- Count the number of movies vs. TV shows.  
- Identify the most common genres and countries producing content.  
- Analyze the number of releases per year to observe trends.  

### **4. Insights & Interpretation**  
- Summarize key findings, such as trends in Netflix's content production over time.  


In [123]:
import pandas as pd
import matplotlib.pyplot as plt

# Load the dataset
dataframe2 = pd.read_csv("netflix_dataset.csv")

# Data Cleaning
newdataframe2 = dataframe2.drop_duplicates().dropna(subset=["title", "listed_in", "release_year"])

newdataframe2["date_added"] = pd.to_datetime(newdataframe2["date_added"], errors="coerce")
newdataframe2["date_added"] = newdataframe2["date_added"].dt.strftime("%B %d, %Y")

# Basic Data Analysis
# Number of Movies vs. TV Shows
type_count = newdataframe2["type"].value_counts()
print("Number of Movies vs. TV Shows")
print(type_count)
print('------------------------------------------------------------------')

# Most Common Genres and Countries Producing Content
genre_count = newdataframe2["listed_in"].str.split(", ").explode().value_counts().head(5)
print("\n\n", "Common Genres")
print(genre_count)
print('------------------------------------------------------------------')

country_count = newdataframe2["country"].str.split(", ").explode().value_counts().head(5)
print("\n\n", "Common Countries Producing Content")
print(country_count)
print('------------------------------------------------------------------')

# Number of Releases per Year
release_count = newdataframe2["release_year"].value_counts().head()
print(release_count)


Number of Movies vs. TV Shows
type
Movie      6131
TV Show    2676
Name: count, dtype: int64
------------------------------------------------------------------


 Common Genres
listed_in
International Movies      2752
Dramas                    2427
Comedies                  1674
International TV Shows    1351
Documentaries              869
Name: count, dtype: int64
------------------------------------------------------------------


 Common Countries Producing Content
country
United States     3689
India             1046
United Kingdom     804
Canada             445
France             393
Name: count, dtype: int64
------------------------------------------------------------------
release_year
2018    1147
2017    1032
2019    1030
2020     953
2016     902
Name: count, dtype: int64


Key Findings:

- There is a higher number of movies created in Netflix than in TV shows.
- The most common genres in Netflix are international movies, dramas, comedies, international TV shows, and documentaries (Top 5 only).
- The most common countries that produce content in Netflix are the United States, India, United Kingdom, Canada, and France (Top 5 only).
- Based on the data, it seems that there are many content released in 2019, 2020, and 2016, wherein 2019 has the most released content with 1030, followed by 953 in 2020 and 902 in 2016.